src/main/java/sg/ncl/DataController.java

```diff
@@ -357,6 +352,53 @@ public String getResources(Model model, @PathVariable String datasetId, HttpSess
         return "data_resources";
     }
 
+    @RequestMapping("{datasetId}/stats")
+    public String getStatistics(Model model,
+                                @PathVariable String datasetId,
+                                @RequestParam(value = "start", required = false) String start,
+                                @RequestParam(value = "end", required = false) String end,
+                                HttpSession session) {
+        if (!validateIfAdmin(session)) {
+            return "nopermission";
+        }
+        if (start == null) {
+            start = "";
+        }
+        if (end == null) {
+            end = "";
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Introduce a new variable instead of reusing the parameter "end". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1226)